src/components/JsonInput.vue

```diff
@@ -28,33 +28,55 @@
   import { useScrollLinking } from '@/compositions'
   import { stringify } from '@/utilities/json'
 
+  type ModelValue = string | number | boolean | undefined | null
+  type ExtendedModelValue = ModelValue | ModelValue[] | Record<string, ModelValue>
+
   const props = defineProps<{
-    modelValue: string | undefined,
+    modelValue: ExtendedModelValue,
```

Comment:
Curious why the more complex model? JSON must be a string so I think its understandable for this component to only accept a string. Love to hear your thoughts on this.